src/app/shared/services/course.service.ts

```diff
@@ -1,65 +1,56 @@
 import { Injectable } from '@angular/core';
-import { Course } from '../interfaces/course.interface.';
+import { HttpClient} from '@angular/common/http'
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
+
+import { HttpCourse } from '../interfaces/http-course.interface';
 
 @Injectable({
   providedIn: 'root'
 })
 export class CourseService {
 
   idCounter = 3;
+  private quanityOfVisibleCourses = 3;
+  private apiUrl = 'http://localhost:3004';
+  inputSubject = new Subject<string>();
+  loadingSubject = new Subject<boolean>();
 
-  constructor() { }
+  constructor(private http: HttpClient) {
+    this.inputSubject = new BehaviorSubject<string>('');
+   }
 
-  courses: Course[] = [{
-    id: 1,
-    title: 'Some Title 1',
-    creationDate: new Date(2023, 5, 1),
-    duration: 200,
-    description: "Some description 1",
-    topRated: false
-  },
-  {
-    id: 2,
-    title: 'Some Title 2',
-    creationDate: new Date(2023, 5, 30),
-    duration: 100,
-    description: "Some description 2",
-    topRated: false
-  },
-  {
-    id: 3,
-    title: 'Some Title 3',
-    creationDate: new Date(2023, 2, 1),
-    duration: 50,
-    description: "Some description 3",
-    topRated: true
-  }]
+  onFetchCourses(): Observable<HttpCourse[]> {
+    return this.http.get<HttpCourse[]>(`${this.apiUrl}/courses?start=0&count=${this.quanityOfVisibleCourses}`)
+  }
 
-  getListCourses(): Course[] {
-    return this.courses;
+  onFetchCoursesWithFilter(textFragment: string): Observable<HttpCourse[]> {
+    return this.http.get<HttpCourse[]>(`${this.apiUrl}/courses?textFragment=${textFragment}`)
   }
 
-  createCourse(course: Course): void {
-    course.id = ++this.idCounter;
-    this.courses.push(course);
+  onLoadAdditionalCourses() {
+    this.quanityOfVisibleCourses += 3;
+    this.onFetchCourses();
   }
 
-  getCourseById(id: number): Course | undefined {
-    return this.courses.find(course => course.id === id);
+  createCourse(course: HttpCourse) {
+    this.http.post<HttpCourse>(`${this.apiUrl}/courses`, course)
+      .subscribe(response => {
```

Comment:
Also - please remove the subscription from the service

Reply:
ok, done